_posts/2017-01-13-school-ratings.md

```diff
@@ -0,0 +1,178 @@
+
+``` python
+# Concatenate column values
+dat['all_grades'] = (dat['D1 Grade'] +
+                     dat['D2 Grade'] +
+                     dat['D3 Grade'] +
+                     dat['D4 Grade'])
+
+# Assign new variable
+dat['d_or_f'] = (dat['all_grades'].str
+                                  .contains('D|F')
+                                  .astype(str)
+                                  .replace({
+                                    'True': 'D or F',
+                                    'False': 'No Ds or Fs'
+                                    })
+                )
+
+dat[['all_grades', 'd_or_f']].head()
+```
+
+<p>
+<div>
+<table border="1" class="data">
+  <thead>
+    <tr style="text-align: right;">
+      <th>all_grades</th>
+      <th>d_or_f</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>BDDC</td>
+      <td>D or F</td>
+    </tr>
+    <tr>
+      <td>CDDC</td>
+      <td>D or F</td>
+    </tr>
+    <tr>
+      <td>DDCF</td>
+      <td>D or F</td>
+    </tr>
+    <tr>
+      <td>BCCD</td>
+      <td>D or F</td>
+    </tr>
+    <tr>
+      <td>BBCC</td>
+      <td>No Ds or Fs</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+</p>
+
+
+Then I could count schools broken down by that binary variable as well as charter and AEA status. Even though it’s example in the official pandas crosstab documentation, I had never before realized I could crosstab more than two variables by passing lists:
+
+
+```python
+pd.crosstab(
+    dat.fail_grade,
+    [dat.is_charter, dat.is_aea]
+)
+```
+
+<p>
+<div>
+<table border="1" class="data">
+  <thead>
+    <tr>
+      <th></th>
+      <th colspan="2" halign="left">Charter</th>
+      <th colspan="2" halign="left">Non-Charter</th>
+    </tr>
+    <tr>
+      <th></th>
+      <th class="num">AEA</th>
+      <th class="num">Non-AEA</th>
+      <th class="num">AEA</th>
+      <th class="num">Non-AEA</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>D or F</th>
+      <td class="num">59</td>
+      <td class="num">293</td>
+      <td class="num">81</td>
+      <td class="num">4535</td>
+    </tr>
+    <tr>
+      <th>No Ds or Fs</th>
+      <td class="num">90</td>
+      <td class="num">247</td>
+      <td class="num">153</td>
+      <td class="num">3167</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+</p>
+
+We wanted to apply this method only to schools in the Austin area. Because charter schools can be members of school districts that are based in other Texas regions, we decided a geospatial query would be the most reliable way to identify schools in the Austin area. I used `fiona` to read shapefiles of Texas schools and education regions, `pyproj` to reproject schools points to the same coordinate system as the regions and shapely to test if schools are located inside the Austin education region:
```

Comment:
reading through i was hoping for a link to [fiona](https://github.com/Toblerity/Fiona) to check it out but that could just be me